core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -245,99 +239,183 @@ private[spark] class CoarseMesosSchedulerBackend(
    */
   override def resourceOffers(d: SchedulerDriver, offers: JList[Offer]) {
     stateLock.synchronized {
-      val filters = Filters.newBuilder().setRefuseSeconds(5).build()
-      for (offer <- offers.asScala) {
+      logDebug(s"Received ${offers.size} resource offers.")
+
+      val (goodOffers, badOffers) = offers.asScala.partition { offer =>
```

Comment:
So looks like a Good and Bad offers are divided simply by checking attributes here, and then we're furthering down filtering more based on other requirements right? how about calling this matchedOffers and unmatchedOffers so we're consistent with our terminology here?